### PR TITLE
KEP-1461 Added swarm_info_esr_url option to allow testers to specify …

### PR DESCRIPTION
**Pull request by rnistuk**
…an Etherium server url that serves the swarm registry contract

### CODE REVIEW

File: bootstrap/bootstrap_peers.cpp [hunk 1]
```diff
@@ -57,12 +57,12 @@ bootstrap_peers::fetch_peers_from_url(const std::string& url)
 
 
 bool
-bootstrap_peers::fetch_peers_from_esr_contract(const std::string &esr_address, const bzn::uuid_t &swarm_id)
+bootstrap_peers::fetch_peers_from_esr_contract(const std::string &esr_url, const std::string &esr_address, const bzn::uuid_t &swarm_id)
```
Comment (ebruck) [inline bootstrap/bootstrap_peers.cpp]: ref with the type

File: bootstrap/bootstrap_peers.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace bzn
 
         bool fetch_peers_from_url(const std::string& url) override;
 
-        bool fetch_peers_from_esr_contract(const std::string &esr_address, const bzn::uuid_t &swarm_id) override;
+        bool fetch_peers_from_esr_contract(const std::string &esr_url, const std::string &esr_address, const bzn::uuid_t &swarm_id) override;
```
Comment (ebruck) [inline bootstrap/bootstrap_peers.hpp]: same

File: swarm/main.cpp [hunk 3]
```diff
@@ -103,7 +103,7 @@ init_logging(const bzn::options& options)
 
 
 bool
-init_peers(bzn::bootstrap_peers& peers, const std::string& peers_file, const std::string& peers_url, const std::string &swarm_info_esr_address, const bzn::uuid_t& swarm_id)
+init_peers(bzn::bootstrap_peers& peers, const std::string& peers_file, const std::string& peers_url, const std::string &swarm_info_esr_url, const std::string &swarm_info_esr_address, const bzn::uuid_t& swarm_id)
```
Comment (ebruck) [inline swarm/main.cpp]: ref with type

File: bootstrap/bootstrap_peers.cpp (hunk 1 above)
Comment (rnistuk) [reply bootstrap/bootstrap_peers.cpp]: Right, fixing

File: bootstrap/bootstrap_peers.hpp (hunk 2 above)
Comment (rnistuk) [reply bootstrap/bootstrap_peers.hpp]: Fixed

File: swarm/main.cpp (hunk 3 above)
Comment (rnistuk) [reply swarm/main.cpp]: Done

**Comment by rnistuk** [comment]
Thanks Ed!

**Pull request merged**